tests/components/smartenergy_awattar/test_config_flow.py

```diff
@@ -0,0 +1,146 @@
+"""Test the Awattar config flow and options flow."""
+
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.smartenergy_awattar.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult, FlowResultType
+
+from tests.common import MockConfigEntry
+
+
+async def _initialize_and_assert_flow(hass: HomeAssistant) -> FlowResult:
+    """Initialize the config flow and do basic checks."""
+    result_init = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result_init["type"] == FlowResultType.FORM
+    assert result_init["errors"] is None
+
+    return result_init
+
+
+async def _initialize_and_assert_options(hass: HomeAssistant, data: dict) -> FlowResult:
+    """Initialize the config flow with options and do basic checks."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN, unique_id="awattar", data=data, entry_id="test"
+    )
+    config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    result_init = await hass.config_entries.options.async_init(config_entry.entry_id)
+
+    assert result_init["type"] == FlowResultType.FORM
+    assert result_init["errors"] is None
+
+    return result_init
+
+
+async def _assert_invalid_scan_interval(
+    flow_id: str,
+    data: dict,
+    configure_fn,
+    err_msg: str,
+) -> None:
+    """Test an error is created when scan interval is invalid."""
+    with pytest.raises(Exception) as exception_info:
+        await configure_fn(
+            flow_id,
+            data,
+        )
+    assert str(exception_info.value) == err_msg
+
+
+async def test_config_flow_init(hass: HomeAssistant, config_1) -> None:
+    """Test we can configure the integration via config flow."""
+    with patch(
+        f"homeassistant.components.{DOMAIN}.state.AwattarApi.get_electricity_price",
```

Comment:
set your domain here, no need to do a fstring, makes it more complicated than it needs to be imo. there's nothing wrong with it I guess, but it is just unneeded.